### PR TITLE
KOGITO-3291: Implement security at Quarkus level for Management Console

### PR DESCRIPTION
**Pull request by nmirasch**
Related jira: https://issues.redhat.com/browse/KOGITO-3291

### CODE REVIEW

**Comment by nmirasch** [comment]
jenkins retest this please

File: security-commons/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  </dependencyManagement>
+  <dependencies>
+
+    <!--dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-vertx-web</artifactId>
+    </dependency -->
+    <!--dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-resteasy</artifactId>
```
Comment (Sgitario) [inline security-commons/pom.xml]: If these dependencies are not necessary, please remove them.

File: ui-packages/packages/common/src/utils/KeycloakClient.ts [hunk 2]
```diff
@@ -31,74 +35,143 @@ const keycloakInstance = createKeycloakInstance();
...
   if (isAuthEnabled()) {
-    getKeycloakInstance()
-      .init({ onLoad: 'login-required' })
-      .success(authenticated => {
-        if (authenticated) {
-          appRender();
-        }
-      });
-
+    if (isReactAuthEnabled()) {
```
Comment (Sgitario) [inline ui-packages/packages/common/src/utils/KeycloakClient.ts]: The name "isReactAuthEnabled" is accurated? Now, the auth is implemented in quarkus, so maybe just "isAuthEnabled" would be more correct?

File: ui-packages/packages/management-console/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "precommit": "lint-staged",
     "build:prod": "yarn run lint && webpack --config webpack.prod.js",
     "start": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js",
-    "start-auth": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js --define process.env.KOGITO_AUTH_ENABLED=true",
+    "start-auth": "webpack-dev-server --hot --color --progress --info=true --config webpack.dev.js --define process.env.KOGITO_AUTH_ENABLED=true --define process.env.KOGITO_REACT_AUTH_ENABLED=true",
```
Comment (Sgitario) [inline ui-packages/packages/management-console/package.json]: Why do we need this property "KOGITO_REACT_AUTH_ENABLED"? Should not we use the KOGITO_AUTH_ENABLED for both quarkus and react apps as they are tied with each other now?

**Review by Sgitario — CHANGES_REQUESTED** [review]
lgtm, but why do we need the KOGITO_REACT_AUTH_ENABLED new property?

File: ui-packages/packages/management-console/package.json (hunk 3 above)
Comment (nmirasch) [reply ui-packages/packages/management-console/package.json]: @Sgitario the idea is to maintain the two modes of security: with the quarkus application (oidc extension) and at the react side (keycloak.js). That reactAuth enabled (property "KOGITO_REACT_AUTH_ENABLED) has the goal to differentiate them.
The common use would be the quarkus security, but I keep this thinking in future integrations of the MC(react side).

Comment (Sgitario) [reply ui-packages/packages/management-console/package.json]: I was not aware of this functionality... Is this new property transparent of the overal usage of the KOGITO_AUTH_ENABLED? I mean if I enable the KOGITO_AUTH_ENABLED property, the security will be enabled without taking care of more properties?

Comment (nmirasch) [reply ui-packages/packages/management-console/package.json]: yes,  when security is enabled.. the property KOGITO_AUTH_ENABLED have to be true in both cases, and if its false is disabled. the only thing is, if the console want to be ran outside the quarkus app with security enabled, an extra property have to be set then :  KOGITO_REACT_AUTH_ENABLED.

Comment (Sgitario) [reply ui-packages/packages/management-console/package.json]: Ok, thanks for the explanation.

**Review by Sgitario — APPROVED** [review]
lgtm

**Comment by nmirasch** [comment]
jenkins retest this please

File: security-commons/.gitignore [hunk 4]
```diff
@@ -0,0 +1,14 @@
+/target
```
Comment (cristianonicolai) [inline security-commons/.gitignore]: this file can probably removed, there is one at the root level already

File: security-commons/src/main/java/org/kie/kogito/security/UserResource.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    protected void setSecurityIdentity(SecurityIdentity securityIdentity) {
+        this.identity = securityIdentity;
+    }
+
+    public static class User {
+
+        private String userName = "Anonymous";
+        private Set<String> roles= Collections.emptySet();
```
Comment (cristianonicolai) [inline security-commons/src/main/java/org/kie/kogito/security/UserResource.java]:
```suggestion
        private Set<String> roles = Collections.emptySet();
```

File: security-commons/src/main/java/org/kie/kogito/security/UserResource.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        private String token = "";
+
+        User(SecurityIdentity identity) {
+            if(identity != null &&
+                    identity.getPrincipal()!=null &&
```
Comment (cristianonicolai) [inline security-commons/src/main/java/org/kie/kogito/security/UserResource.java]:
```suggestion
                    identity.getPrincipal() != null &&
```

File: security-commons/src/main/java/org/kie/kogito/security/UserResource.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                    identity.getCredential(TokenCredential.class)!=null) {
```
Comment (cristianonicolai) [inline security-commons/src/main/java/org/kie/kogito/security/UserResource.java]:
```suggestion
                    identity.getCredential(TokenCredential.class) != null) {
```

**Review by cristianonicolai — COMMENTED** [review]
@nmirasch overall looks good to me, just missing two things I guess:
- Add the new security-commons module to kogito-bom just to be on the safe side for the managed version.
- Also, not sure if I missed but I would be good to have a IT test to check the endpoint when running the managent console app.

**Comment by nmirasch** [comment]
depends on https://github.com/kiegroup/kogito-runtimes/pull/797

**Comment by nmirasch** [comment]
@pefernan, @cristianonicolai @Sgitario  I have added the security-commons module to kogito-bom.. now this PR depends on  https://github.com/kiegroup/kogito-runtimes/pull/797. I have removed the security implementation at react part. Could you take a look please?

**Comment by pefernan** [comment]
jenkins retest this please

**Review by pefernan — APPROVED** [review]
+1 with last changes looks and works great. Thanks!

**Review by cristianonicolai — APPROVED** [review]
great work @nmirasch !

**Comment by pefernan** [comment]
jenkins retest this

**Review by jstastny-cz — APPROVED** [review]
LGTM

**Pull request merged**